### PR TITLE
PoW mining - deprecation

### PR DESCRIPTION
**Pull request by macfarla**
## PR description
Deprecation notices in help; log message if mining is enabled; `@Deprecation` annotations on mining RPC methods

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java [hunk 1]
```diff
@@ -52,27 +52,34 @@
...
 public class MiningOptions implements CLIOptions<MiningConfiguration> {
 
+  private static final String DEPRECATION_PREFIX =
+      "Deprecated. PoW consensus is deprecated. See CHANGELOG for alternative options. ";
+
   @Option(
       names = {"--miner-enabled"},
-      description = "Set if node will perform mining (default: ${DEFAULT-VALUE})")
+      description =
+          DEPRECATION_PREFIX + "Set if node will perform mining (default: ${DEFAULT-VALUE})")
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java:61]: this is used for QBFT so may need a rename rather than deprecation

Comment (siladu) [reply besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java:61]: I was wrong, it isn't used for QBFT actually

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java [hunk 2]
```diff
@@ -41,6 +41,7 @@
 import com.google.common.base.Suppliers;
 import org.apache.tuweni.units.bigints.BaseUInt256Value;
 
+@Deprecated(since = "24.11.0")
```
Comment (siladu) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java:44]: More likely to be 24.12.0

File: besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java (hunk 1 above)
Comment (jframe) [reply besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java:61]: Yep QBFT doesn't use the start/stop CLI options they aren't ignored so it's fine to deprecate this.

Comment (macfarla) [reply besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java:61]: awesome, thanks both for confirming

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java (hunk 2 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java]: changed everywhere

**Review by fab-10 — APPROVED** [review]

**Pull request merged**